### PR TITLE
refactor: update `blas/ext/base/dapxsum` to follow current project conventions

### LINKED ISSUE #1465: [RFC]: refactor `blas/ext/base/dapxsum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dapxsum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dapxsum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by nightknighto**
Resolves #1465 .

## Description

This pull request refactors `blas/ext/base/dapxsum` to follow current project conventions.

## Related Issues

Related issues #788 , #1152

This pull request:

- resolves #1465 
- progresses #1152 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md [hunk 1]
```diff
@@ -56,17 +56,15 @@ The function has the following parameters:
 -   **x**: input [`Float64Array`][@stdlib/array/float64].
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md:57]:
```suggestion
-   **alpha**: scalar constant.
-   **x**: input [`Float64Array`][@stdlib/array/float64].
```

Missing parameter which we can go ahead and fix in this PR.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md [hunk 2]
```diff
@@ -75,15 +73,12 @@ Note that indexing is relative to the first index. To introduce an offset, use [
...
 
 var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
 var x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
 
-var N = floor( x0.length / 2 );
-
-var v = dapxsum( N, 5.0, x1, 2 );
-// returns 25.0
+var v = dapxsum( 3, 5.0, x1, 2 );
+// returns 16
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md]:
```suggestion
var v = dapxsum( 4, 5.0, x1, 2 );
// returns 25.0
```

There's no reason to change the length here.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md [hunk 3]
```diff
@@ -104,17 +99,15 @@ The function has the following additional parameters:
...
 var Float64Array = require( '@stdlib/array/float64' );
-var floor = require( '@stdlib/math/base/special/floor' );
 
 var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
-var N = floor( x.length / 2 );
 
-var v = dapxsum.ndarray( N, 5.0, x, 2, 1 );
-// returns 25.0
+var v = dapxsum.ndarray( 3, 5.0, x, 2, 1 );
+// returns 16
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md]:
```suggestion
var v = dapxsum.ndarray( 4, 5.0, x, 2, 1 );
// returns 25.0
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md [hunk 4]
```diff
@@ -138,18 +131,11 @@ var v = dapxsum.ndarray( N, 5.0, x, 2, 1 );
...
 var dapxsum = require( '@stdlib/blas/ext/base/dapxsum' );
 
-var x;
-var i;
-
-x = new Float64Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float64', discreteUniform( 0, 100 ) );
```

This package is explicitly intended for double-precision floating-point strided arrays.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md [hunk 5]
```diff
@@ -53,20 +53,19 @@ var v = dapxsum( N, 5.0, x, 1 );
...
 var Float64Array = require( '@stdlib/array/float64' );
-var floor = require( '@stdlib/math/base/special/floor' );
 
 var x = new Float64Array( [ 1.0, 2.0, 2.0, -7.0, -2.0, 3.0, 4.0, 2.0 ] );
-var N = floor( x.length / 2 );
 
-var v = dapxsum( N, 5.0, x, 2 );
-// returns 25.0
+var v = dapxsum( 3, 5.0, x, 2 );
+// returns 16
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/README.md]:
```suggestion
var v = dapxsum( 4, 5.0, x, 2 );
// returns 25.0
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt [hunk 6]
```diff
@@ -37,27 +37,26 @@
     > {{alias}}( x.length, 5.0, x, 1 )
     16.0
 
-    // Using `N` and `stride` parameters:
+    // Using `N` and stride parameters:
     > x = new {{alias:@stdlib/array/float64}}( [ -2.0, 1.0, 1.0, -5.0, 2.0, -1.0 ] );
-    > var N = {{alias:@stdlib/math/base/special/floor}}( x.length / 2 );
     > var stride = 2;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt [hunk 7]
```diff
@@ -37,27 +37,26 @@
...
-    > {{alias}}( N, 5.0, x, stride )
+    > {{alias}}( 3, 5.0, x, stride )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt]:
```suggestion
    > {{alias}}( 3, 5.0, x, 2 )
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt [hunk 8]
```diff
@@ -37,27 +37,26 @@
...
     16.0
 
     // Using view offsets:
     > var x0 = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
     > var x1 = new {{alias:@stdlib/array/float64}}( x0.buffer, x0.BYTES_PER_ELEMENT*1 );
-    > N = {{alias:@stdlib/math/base/special/floor}}( x0.length / 2 );
     > stride = 2;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt [hunk 9]
```diff
@@ -37,27 +37,26 @@
...
-    > {{alias}}( N, 5.0, x1, stride )
+    > {{alias}}( 3, 5.0, x1, stride )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt]:
```suggestion
    > {{alias}}( 3, 5.0, x1, 2 )
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt [hunk 10]
```diff
@@ -91,10 +88,8 @@
 
     // Using offset parameter:
     > var x = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
-    > var N = {{alias:@stdlib/math/base/special/floor}}( x.length / 2 );
-    > {{alias}}.ndarray( N, 5.0, x, 2, 1 )
+    > {{alias}}.ndarray( 3, 5.0, x, 2, 1 )
     14.0
 
     See Also
     --------
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt]:
```suggestion
    --------

```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt [hunk 11]
```diff
@@ -91,8 +88,7 @@
 
     // Using offset parameter:
     > var x = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/docs/repl.txt]:
```suggestion
    > x = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/examples/index.js [hunk 12]
```diff
@@ -18,18 +18,11 @@
...
 var dapxsum = require( './../lib' );
 
-var x;
-var i;
-
-x = new Float64Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float64', discreteUniform( 0, 100 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/lib/ndarray.native.js [hunk 13]
```diff
@@ -48,10 +49,11 @@ var addon = require( './dapxsum.native.js' );
 */
 function dapxsum( N, alpha, x, stride, offset ) {
 	var view;
+	offset = minViewBufferIndex( N, stride, offset );
 	if ( stride < 0 ) {
 		offset += (N-1) * stride;
 	}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/lib/ndarray.native.js]:
```suggestion
```

No longer needed.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/manifest.json [hunk 14]
```diff
@@ -35,6 +35,10 @@
 			],
 			"libpath": [],
 			"dependencies": [
+        "@stdlib/napi/export",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/manifest.json:39]: This file has mixed TABS and spaces. The indentation should be 2-space.

This file is also missing separate build configurations for benchmarks and examples. See, e.g., https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/dapxsumors/manifest.json.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/manifest.json [hunk 15]
```diff
@@ -35,6 +35,10 @@
...
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-float",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/manifest.json]:
```suggestion
        "@stdlib/napi/argv-double",
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/dapxsum.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_strided_float32array.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c]:
```suggestion
#include "stdlib/napi/argv_strided_float64array.h"
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/dapxsum.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c:22]:
```suggestion
#include "stdlib/napi/argv_int64.h"
#include "stdlib/napi/argv_double.h"
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 1 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
	STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 2 );
```

You didn't extract the scalar constant argument.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this @Deadreyo. Left a few comments. I also believe that there are missing changes to test files and some other files in `lib`. E.g., see this recent PR: https://github.com/stdlib-js/stdlib/pull/1993/files. Once resolved, this PR should be good to move forward.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/manifest.json (hunk 14 above)
Comment (nightknighto) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsum/manifest.json:39]: Added the missing confs.
For the indentation thing, I've run the document formatting tool on the file and they all look 2-spaces indented. If still not the correct result, please let me know how to deal with it

**Comment by nightknighto** [comment]
Updated missing changes in `lib` and `test` files

**Review by Planeshifter — APPROVED** [review]
Did a review and the changes look all good to me. Thank you for working on this, @Deadreyo.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 2 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsum/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, stride, argv, 2 );
```

**Pull request merged**